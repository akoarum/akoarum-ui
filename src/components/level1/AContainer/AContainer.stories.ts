import { storiesOf } from '@storybook/vue'
import { action } from '@storybook/addon-actions'
import { withKnobs, text, color } from '@storybook/addon-knobs'
import { withInfo } from 'storybook-addon-vue-info'
import AContainer from './AContainer.vue'

storiesOf('level1/AContainer', module)
  .addDecorator(withInfo)
  .addDecorator(withKnobs)
  .add('Normal', () => ({
    components: { AContainer },
    props: {
      tag: { default: text('要素', 'div') },
      padding: { default: text('padding', '') },
      background: { default: color('背景色', '#fff') }
    },
    template: `
      <AContainer :tag="tag" :padding="padding" :background="background">
        <p>テキスト</p>
      </AContainer>
    `
  }), { info: {} })
