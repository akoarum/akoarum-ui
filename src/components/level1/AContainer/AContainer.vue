<script lang="ts">
import { CreateElement, VNode } from 'vue'
import { defineComponent, computed } from '@vue/composition-api'

interface Props {
  tag: string
  padding: string
  background?: string
}

export default defineComponent({
  props: {
    tag: { type: String, default: 'div' },
    padding: { type: String },
    background: { type: String }
  },
  setup(props: Props) {
    const styled = computed<{ [key: string]: string }>(() => {
      const styles: { [key: string]: string } = {}
      if (props.padding) styles['--padding'] = props.padding
      if (props.background) styles['--background'] = props.background
      return styles
    })

    return { styled }
  },
  render(createElement: CreateElement): VNode {
    return createElement(
      this.tag,
      {
        class: 'aContainer',
        style: this.styled
      },
      this.$slots.default
    )
  }
})
</script>

<style lang="scss" scoped>
.aContainer {
  --padding: 14px 16px;
  --background: #fff;
  box-shadow: 0 2px 10px 0 rgba(0, 0, 0, 0.14);
  border-radius: 2px;
  padding: var(--padding);
  background: var(--background);
}
</style>
