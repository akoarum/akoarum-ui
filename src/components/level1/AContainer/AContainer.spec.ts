import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueCompositionApi from '@vue/composition-api'
import AContainer from './AContainer.vue'

describe('AContainer', () => {
  const localVue = createLocalVue()
  let wrapper

  localVue.use(VueCompositionApi)

  beforeEach(() => {
    wrapper = shallowMount(AContainer, {
      propsData: {},
      slots: {
        default: 'テキスト'
      },
      localVue
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })

    it('styled: padding、backgroundはpropsから設定できる', () => {
      wrapper.setProps({ padding: '10px', background: '#f0f0f0' })
      expect(wrapper.vm.styled).toEqual({
        '--padding': '10px',
        '--background': '#f0f0f0'
      })
    })
  })
})
