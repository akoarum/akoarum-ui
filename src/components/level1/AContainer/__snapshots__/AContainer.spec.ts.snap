// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AContainer initialized snapshot 1`] = `
<div
  class="aContainer"
>
  テキスト
</div>
`;
